testsuite/MDAnalysisTests/formats/test_libmdaxdr.py

```diff
@@ -21,376 +21,469 @@
-    assert_equal(prec, 1000.0)
+@pytest.mark.parametrize("xdrfile, fname", ((XTCFile, XTC_multi_frame),
+                                            (TRRFile, TRR_multi_frame)))
+def test_time(xdrfile, fname):
+    with xdrfile(fname) as f:
+        for i, frame in enumerate(f):
+            assert frame.time == i * .5
 
 
-def test_read_multi_frame_xtc():
-    with XTCFile(XTC_multi_frame) as f:
-        f.seek(5)
-        assert_equal(f.tell(), 5)
-        assert_equal(len(f), 10)
-        assert_equal(f.tell(), 5)
-        f.seek(0)
-        assert_equal(f.tell(), 0)
-        ones = np.ones(30).reshape(10, 3)
-        box_compare = np.eye(3) * 20
-        for i, frame in enumerate(f):
-            xyz, box, step, time, prec = frame
-            assert_array_almost_equal(xyz, ones * i, decimal=3)
-            assert_array_almost_equal(box, box_compare, decimal=3)
-            assert_equal(step, i)
-            assert_equal(time, i * 0.5)
-            assert_equal(prec, 1000.0)
-
-
-@run_in_tempdir()
-def test_write_xtc():
-    with XTCFile(XTC_multi_frame) as f_in, XTCFile('foo.xtc', 'w') as f_out:
-        assert_equal(f_out.n_atoms, 0)
-        for frame in f_in:
-            f_out.write(*frame)
-
-    with XTCFile('foo.xtc') as f:
-        assert_equal(len(f), 10)
-        ones = np.ones(30).reshape(10, 3)
-        box_compare = np.eye(3) * 20
-        for i, frame in enumerate(f):
-            xyz, box, step, time, prec = frame
-            assert_array_almost_equal(xyz, ones * i, decimal=3)
-            assert_array_almost_equal(box, box_compare, decimal=3)
-            assert_equal(step, i)
-            assert_equal(time, i * 0.5)
-            assert_equal(prec, 1000.0)
+def test_box_xtc(xtc):
+    box = np.eye(3) * 20
+    for frame in xtc:
+        assert_array_almost_equal(frame.box, box, decimal=3)
+
+
+def test_xyz_xtc(xtc):
+    ones = np.ones(30).reshape(10, 3)
+    for i, frame in enumerate(xtc):
+        assert_array_almost_equal(frame.x, ones * i, decimal=3)
+
+
+def test_box_trr(trr):
+    box = np.eye(3) * 20
+    for frame in trr:
+        assert_array_almost_equal(frame.box, box)
+
+
+def test_xyz_trr(trr):
+    ones = np.ones(30).reshape(10, 3)
+    for i, frame in enumerate(trr):
+        assert_array_almost_equal(frame.x, ones * i)
+
+
+def test_velocities_trr(trr):
+    ones = np.ones(30).reshape(10, 3)
+    for i, frame in enumerate(trr):
+        assert_array_almost_equal(frame.v, ones * i + 10)
+
+
+def test_forces_trr(trr):
+    ones = np.ones(30).reshape(10, 3)
+    for i, frame in enumerate(trr):
+        assert_array_almost_equal(frame.f, ones * i + 20)
+
+
+def test_lmbda_trr(trr):
+    for i, frame in enumerate(trr):
+        assert_almost_equal(frame.lmbda, .01 * i)
+
+
+@pytest.fixture
+def written_xtc(tmpdir, xtc):
+    fname = str(tmpdir.join("foo.xtc"))
+    with XTCFile(fname, 'w') as f:
+        for frame in xtc:
+            f.write(*frame)
+    with XTCFile(fname) as f:
+        yield f
```

Comment:
these fixtures with yield are really nice. They do remove a hell of a lot of boilerplate code